File: src/cxx-builder/smart-solution.cpp
/*

For users: 

This is smart solution (uses greedy aproach).

You should write code as usual(cin, cout),
no need to work with any threads,
all the hard work is done for you

*/

#include <iostream>
#include <algorithm>
using namespace std;

const int INF = 1e9;
const int MAXN = 10;
const int MAXM = 1000;
int n;
int a[MAXN];

void solve() {
    int m;
    cin >> n >> m;
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }

    sort(a, a + n);
    reverse(a, a + n);

    int ans = 0;
    for (int i = 0; i < n; i++) {
        while (m - a[i] >= 0) {
            m -= a[i];
            ans++;
        }
    }

    if (m > 0) {
        cout << -1 << '\n';
    } else {
        cout << ans << '\n';
    }
}

int main() {
    ios_base::sync_with_stdio(0); cin.tie(0);
    solve();
}

File: src/runners/java-runner/smart-solution.java
/*

For users: 

This is smart solution (uses greedy approach).

You should write code as usual(System.in, System.out),
no need to work with any threads,
all the hard work is done for you

*/

import java.util.*;

public class SmartSolution {
    
    private static final int INF = 1000000000;
    private static final int MAXN = 10;
    private static final int MAXM = 1000;
    private static int n;
    private static int[] a = new int[MAXN];
    
    private static void solve() {
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();
        int m = sc.nextInt();
        for (int i = 0; i < n; i++) {
            a[i] = sc.nextInt();
        }
        
        Arrays.sort(a, 0, n);
        // Reverse the array
        for (int i = 0; i < n / 2; i++) {
            int temp = a[i];
            a[i] = a[n - 1 - i];
            a[n - 1 - i] = temp;
        }
        
        int ans = 0;
        for (int i = 0; i < n; i++) {
            while (m - a[i] >= 0) {
                m -= a[i];
                ans++;
            }
        }
        
        if (m > 0) {
            System.out.println(-1);
        } else {
            System.out.println(ans);
        }
        sc.close();
    }
    
    public static void main(String[] args) {
        solve();
    }
}
